src/tactic/simps.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2019 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import tactic.core
+/-!
+  # simps attribute
+  This file defines the `@[simps]` attribute, to automatically generate simp-lemmas
+  reducing a definition when projections are applied to it.
+
+  ## Tags
+  structures, projections, simp, simplifier, generates declarations
+-/
+
+open tactic expr
+
+/-- Add a lemma with `nm` stating that `lhs = rhs`. `type` is the type of both `lhs` and `rhs`,
+  `args` is the list of local constants occurring, and `univs` is the list of universe variables.
+  If `add_simp` then we make the resulting lemma a simp-lemma. -/
+meta def add_projection (nm : name) (type lhs rhs : expr) (args : list expr)
+  (univs : list name) (add_simp : bool) : tactic unit := do
+  eq_ap ← mk_app `eq [type, lhs, rhs],
+  refl_ap ← mk_app `eq.refl [type, lhs],
+  let decl_type := eq_ap.pis args,
+  let decl_value := refl_ap.lambdas args,
+  let decl := declaration.thm nm univs decl_type (pure decl_value),
+  add_decl decl <|> fail format!"failed to add projection lemma {nm}.",
+  when add_simp $ set_basic_attribute `simp nm tt
```

Comment:
This change fixed the `rfl`-lemma issue for me:
```suggestion
  when add_simp $ set_basic_attribute `simp nm tt >> set_basic_attribute `_refl_lemma nm tt
```
As far as I can see, it should always be correct to do so.

Reply:
What is this attribute? Is it for `dsimp`? I don't see any occurrence of it in mathlib.

Reply:
Lean automatically adds it when the definition of a `simp` lemma is exactly `rfl`, and it is used for `dsimp`, yes.

Reply:
Oh, I only now see your previous messages! I will add this.